.gitignore

```diff
@@ -115,7 +116,13 @@ hs_err_pid*
 
 ### Gradle ###
 .gradle
-build/
+build/classes
+build/generated
+build/reports
+build/test-results
+build/tmp
+build/.DS_Store
+build/libs/.DS_Store
```

Comment:
If you need to include `build/libs` but exclude everything else, I think you can do:
```
**/build/**
!build/libs
```

Likewise you can globally ignore DS_Store with `**/DS_Store`

Reply:
Ahhh fancy!